wordpress-mobile/WordPress-Android
# Issue/11133 update page item labels

Fixes #11133 

**Review instructions:**
1. Review and merge https://github.com/wordpress-mobile/WordPress-Android/pull/11257
3. Merge `feature/master-pages-offline-support` into `unhandled_autosave`
1. Review this PR
2. Wait until https://github.com/wordpress-mobile/WordPress-Android/pull/11228 gets merged
3. Update the target branch with `feature/master-pages-offline-support`
4. Remove the "Not ready for merge" label
4. Merge this PR

Adds the following labels to page list items. The size of the PR seems bigger than it actually is. Most of the code was copy-pasted from post list.

**Page state**
- Pending review
- Local draft
- Local changes
- Private
- Uploading page
- Uploading draft
- Queued page

**Changes are enqueued for auto-upload**
- We'll publish the page when your device is back online.
- We'll submit your page for review when your device is back online.
- We'll schedule your page when your device is back online.
- We'll publish your private page when your device is back online.
- We'll save your draft when your device is back online

**Auto-upload failed but the changes are still enqueued for auto-upload**
- We couldn't publish this page, but we'll try again later.
- We couldn't publish this private page, but we'll try again later.
- We couldn't schedule this page, but we'll try again later.
- We couldn't submit this page for review, but we'll try again later. 

**Auto-upload failed multiple times and the changes were removed from the queue**
- We couldn't upload this media, and didn't publish the page.
- We couldn't upload this media, and didn't publish this private page.
- We couldn't upload this media, and didn't schedule this page.
- We couldn't upload this media, and didn't submit this page for review. 


I decided to introduce unit tests in another PR - https://github.com/wordpress-mobile/WordPress-Android/pull/11281.

To test:
The business logic was copy pasted from PostListItemUiStateHelper. I'd suggest testing just a couple of scenarios. Also please make sure to review all the changes in strings.xml - I tried to replace all mentions of `post` with `page` but I might have missed something.
We'll test all the scenarios before we merge the master branch into develop.

PR submission checklist:
